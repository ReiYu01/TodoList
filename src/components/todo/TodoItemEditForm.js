import React, { useState } from 'react'
import { connect } from 'react-redux'
import { changeText, toggleTodo } from '../../actions/'

function TodoItemEditForm(props) {
  const [editText, setEditText] = useState(props.value.text)

  //console.log(props)

  // 先解構賦值，直接套用由props得到的變數值
  const { value, changeText, toggleTodo } = props

  const onChange = (event) => {
    setEditText(event.target.value)
  }

  const handleEditSave = (id, text) => {
    toggleTodo(id)
    changeText(id, text)
  }

  return (
    <>
      <li className="list-group-item d-flex justify-content-between align-items-center">
        <div className="form-inline">
          <input
            id="todoEdit"
            className="form-control mb2"
            type="text"
            value={editText}
            onChange={onChange}
          />
          <button
            type="button"
            className="btn btn-info mb2"
            onClick={() => {
              handleEditSave(value.id, editText)
            }}
          >
            save
          </button>
        </div>
      </li>
    </>
  )
}

const mapStateToProps = null

const mapDispatchToProps = {
  toggleTodo,
  changeText,
}

export default connect(mapStateToProps, mapDispatchToProps)(TodoItemEditForm)
