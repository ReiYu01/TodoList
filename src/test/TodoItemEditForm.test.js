import React from 'react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import TodoAddForm from '../components/todo/TodoAddForm'
import Enzyme, { mount, shallow, render } from 'enzyme'
import setupTest from './setupTests'
import TodoItemEditForm from '../components/todo/TodoItemEditForm'

describe('Should Render TodoItemEditForm', () => {
  const mockStore = configureStore()
  const store = mockStore()

  const value = {
    id: 678678678,
    text: '每週三交進度',
    completed: false,
    edited: false,
  }

  //====render出畫面，檢查HTML====
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<TodoAddForm store={store} value={value} debug />)

    expect(component).toMatchSnapshot()
  })
})
