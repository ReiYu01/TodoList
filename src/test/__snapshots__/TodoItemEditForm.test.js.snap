// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render TodoItemEditForm renders correctly 1`] = `
<li
  className="list-group-item d-flex justify-content-between align-items-center"
>
  <div
    className="form-inline"
  >
    <input
      className="form-control mb2"
      id="todoEdit"
      onChange={[Function]}
      type="text"
      value="每週三交進度"
    />
    <button
      className="btn btn-info mb2"
      onClick={[Function]}
      type="button"
    >
      save
    </button>
  </div>
</li>
`;

exports[`Should Render TodoItemEditForm should render correctly in "debug" mode 1`] = `
<TodoAddForm
  addTodo={[Function]}
  debug={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  value={
    Object {
      "completed": false,
      "edited": false,
      "id": 678678678,
      "text": "每週三交進度",
    }
  }
/>
`;
